src/main.js

```diff
@@ -47,51 +48,79 @@ function checkedToString(value) {
+  { options: { flagProtection, interpolation, shell }, process: { env } },
   { getDefaultShell, getShellName }
 ) {
+  flagProtection = flagProtection ? true : false;
   interpolation = interpolation ? true : false;
   shell = isString(shell) ? shell : getDefaultShell({ env });
 
   const shellName = getShellName({ shell }, { resolveExecutable });
-  return { interpolation, shellName };
+  return { flagProtection, interpolation, shellName };
 }
 
 /**
  * Escapes an argument for the given shell.
  *
  * @param {object} args The arguments for this function.
  * @param {string} args.arg The argument to escape.
+ * @param {boolean} args.flagProtection Is flag protection enabled.
  * @param {boolean} args.interpolation Is interpolation enabled.
  * @param {string} args.shellName The name of the shell to escape `arg` for.
  * @param {object} deps The dependencies for this function.
  * @param {Function} deps.getEscapeFunction Get the escape function for a shell.
+ * @param {Function} deps.stripFlagPrefix A function to strip flag prefixes.
  * @returns {string} The escaped argument.
  * @throws {TypeError} The argument to escape is not stringable.
  */
-function escape({ arg, interpolation, shellName }, { getEscapeFunction }) {
+function escape(
+  { arg, flagProtection, interpolation, shellName },
+  { getEscapeFunction, stripFlagPrefix }
+) {
   const argAsString = checkedToString(arg);
   const escape = getEscapeFunction(shellName, { interpolation });
   const escapedArg = escape(argAsString);
-  return escapedArg;
+  if (flagProtection) {
+    return stripFlagPrefix(escapedArg);
+  } else {
+    return escapedArg;
+  }
 }
 
 /**
  * Quotes and escape an argument for the given shell.
  *
  * @param {object} args The arguments for this function.
  * @param {string} args.arg The argument to escape.
+ * @param {boolean} args.flagProtection Is flag protection enabled.
  * @param {string} args.shellName The name of the shell to escape `arg` for.
  * @param {object} deps The dependencies for this function.
  * @param {Function} deps.getQuoteFunction Get the quote function for a shell.
+ * @param {Function} deps.stripFlagPrefix A function to strip flag prefixes.
  * @returns {string} The quoted and escaped argument.
  * @throws {TypeError} The argument to escape is not stringable.
  */
-function quote({ arg, shellName }, { getQuoteFunction }) {
+function quote(
+  { arg, flagProtection, shellName },
+  { getQuoteFunction, stripFlagPrefix }
+) {
   const argAsString = checkedToString(arg);
   const quote = getQuoteFunction(shellName);
   const escapedAndQuotedArg = quote(argAsString);
-  return escapedAndQuotedArg;
+  if (flagProtection) {
+    return (
+      escapedAndQuotedArg.substring(0, 1) +
+      stripFlagPrefix(
+        escapedAndQuotedArg.substring(1, escapedAndQuotedArg.length - 1)
+      ) +
+      escapedAndQuotedArg.substring(
+        escapedAndQuotedArg.length - 1,
+        escapedAndQuotedArg.length
+      )
+    );
```

Comment:
This MUST be changed before this is merged.

Not only is this _just objectively_ ugly code, it also creates a coupling between this and the `quote` function that shouldn't be there: (while unlikely to not hold true) it shouldn't be assumed that quoting adds 1 character at the start and one character at the end.